pennylane_qiskit/vqe_runtime_runner.py

```diff
@@ -0,0 +1,488 @@
+        wires_map = OrderedDict(zip(all_wires, consecutive_wires))
+
+        # From here: Create the Qiskit ansatz circuit
+        params_vector = ParameterVector("p", num_params)
+
+        reg = QuantumRegister(num_qubits, "q")
+        circuit_ansatz = QuantumCircuit(reg, name="vqe")
+
+        circuits = []
+
+        j = 0
+        for operation in tape.operations:
+            wires = operation.wires.map(wires_map)
+            par = operation.parameters
+            operation = operation.name
+            mapped_operation = QiskitDevice._operation_map[operation]
+
+            qregs = [reg[i] for i in wires.labels]
+
+            if operation.split(".inv")[0] in ("QubitUnitary", "QubitStateVector"):
+                # Need to revert the order of the quantum registers used in
+                # Qiskit such that it matches the PennyLane ordering
+                qregs = list(reversed(qregs))
+
+            dag = circuit_to_dag(QuantumCircuit(reg, name=""))
+
+            if operation in ("QubitUnitary", "QubitStateVector"):
+                # Parameters are matrices
+                gate = mapped_operation(par[0])
+            else:
+                # Parameters for the operation
+                if par:
+                    # Trainable parameter
+                    if qml.math.requires_grad(par[0]):
+                        op_num_params = len(par)
+                        par = []
+                        for num in range(op_num_params):
+                            par.append(params_vector[j + num])
+                        j += op_num_params
+
+                gate = mapped_operation(*par)
+
+            if operation.endswith(".inv"):
+                gate = gate.inverse()
+
+            dag.apply_operation_back(gate, qargs=qregs)
+            circuit = dag_to_circuit(dag)
+            circuits.append(circuit)
+
+        for circuit in circuits:
+            circuit_ansatz &= circuit
+
+    else:
+        raise ValueError("Input ansatz is not a quantum function or a string.")
+
+    return x0, circuit_ansatz, num_qubits, all_wires
+
+
+def hamiltonian_to_list_string(hamiltonian, wires):
+    r"""Convert a Hamiltonian object from PennyLane to a list of pairs representing each coefficient and
+    term in the Hamiltonian.
+
+    Args:
+        hamiltonian (qml.Hamiltonian): A Hamiltonian from PennyLane.
+        wires (qml.wires.Wires): A list of qubits from PennyLane.
+
+    Returns:
+        list[tuple[float,str]]: Hamiltonian in a format for the runtime program.
+    """
+
+    num_qubits = len(wires)
+
+    consecutive_wires = qml.wires.Wires(range(num_qubits))
+    wires_map = OrderedDict(zip(wires, consecutive_wires))
+
+    coeff, observables = hamiltonian.terms
+
+    authorized_obs = {"PauliX", "PauliY", "PauliZ", "Hadamard", "Identity"}
+
+    for obs in observables:
+        # Tensors
+        if isinstance(obs.name, list):
+            for ob in obs.name:
+                if ob not in authorized_obs:
```

Comment:
This should evaluate in a lazy fashion and could reduce the complexity of the function.
```suggestion
            if any(ob not in authorized_obs for ob in obs.name):
```